### PR TITLE
[QoL] Improve readability in MovePhase

### PR DESCRIPTION
**Pull request by mcmontag**
## What are the changes?
Nothing functional, at least not intentionally.  Added documentation and cleaned up `MovePhase`.

## Why am I doing these changes?
It's extremely difficult to follow in its current state.  Adding documentation and cleaning up the code here (and soon `MoveEffectPhase`) will pay off in the short term, and will ultimately make it easier to game-plan a larger refactor.

## What did change?
Large sections of code were split out into methods and variables were renamed for readability; quite a few comments were added in places where I felt they'd be helpful.  A few things were shuffled around, but only in non-functional ways (e.g. moving a variable declaration closer to where it ends up being used).

### Screenshots/Videos
n/a

## How to test the changes?
If you find a section of shuffled code that seems like it may affect a move, test it.

All tests are passing, without any need to update them.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/phases.ts [hunk 1]
```diff
@@ -2565,6 +2565,10 @@ export class MovePhase extends BattlePhase {
   protected failed: boolean;
   protected cancelled: boolean;
 
+  /**
+   * @param followUp Indicates that the move being uses is a "follow-up" - for example, a move being used by Metronome or Dancer.
+   *                 Follow-ups bypass a few failure conditions, including flinches, sleep/paralysis/freeze checks, and
+   */
   constructor(scene: BattleScene, pokemon: Pokemon, targets: BattlerIndex[], move: PokemonMove, followUp?: boolean, ignorePp?: boolean) {
```
Comment (mcmontag) [inline src/phases.ts:2572]: `followUp` is the only param that isn't completely obvious, so it's the only one I called out.

**Comment by flx-sta** [comment]
@mcmontag since this is a refactoring I have to ask you to make sure that tests are written to cover that the results are the same as before.

**Comment by mcmontag** [comment]
@flx-sta makes sense. I'll have to take some time to think about how I want to go about getting full coverage.

I'll mark this as a draft for now!

**Comment by DayKev** [comment]
> @mcmontag since this is a refactoring I have to ask you to make sure that tests are written to cover that the results are the same as before.

Tbh I think the existing tests should be sufficient; it's a refactor of `MovePhase` so if he breaks something then at least one of the existing move tests should break as well.

**Pull request closed without merging**